lib/matplotlib/pyplot.py

```diff
@@ -1149,7 +1149,7 @@ def subplots(nrows=1, ncols=1, sharex=False, sharey=False, squeeze=True,
     return fig, axs
 
 
-def subplot2grid(shape, loc, rowspan=1, colspan=1, **kwargs):
+def subplot2grid(shape, loc, rowspan=1, colspan=1, fig=gcf(), **kwargs):
```

Comment:
I know this is closed, but an important fact about python is that the default values for function args are evaluated at definition time (and never again).  Hence, this will call `gcf()` once when this function in defined and then will always default to _that_ figure in the future (not what ever the current 'current figure' is).